src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -1253,5 +1256,56 @@ CHIP_ERROR VerifyCertificateSigningRequestFormat(const uint8_t * csr, size_t csr
     return CHIP_NO_ERROR;
 }
 
+const char * PemEncoder::NextLine()
+{
+    bool hasLine = false;
+
+    switch (mState)
+    {
+    case State::kPrintHeader: {
+        // The `.48s` is to make sure the header is not wider than out internal string buffer. We clamp the header.
+        mStringBuilder.Reset().AddFormat("-----BEGIN %.48s-----", mEncodedElement);
+        mState  = mDerBytes.empty() ? State::kPrintFooter : State::kPrintBody;
+        hasLine = true;
+        break;
+    }
+    case State::kPrintBody: {
+        size_t remaining      = mDerBytes.size() - mProcessedBytes;
+        size_t chunkSizeBytes = std::min(remaining, kNumBytesPerLine);
+
+        {
+            char base64EncodedBuf[kLineBufferSize];
+            size_t encodedLen = static_cast<size_t>(
+                Base64Encode(mDerBytes.data() + mProcessedBytes, static_cast<uint16_t>(chunkSizeBytes), base64EncodedBuf));
+            VerifyOrDie(encodedLen < sizeof(base64EncodedBuf));
+            base64EncodedBuf[encodedLen] = '\0';
+            mStringBuilder.Reset().Add(base64EncodedBuf);
+        }
+
+        mProcessedBytes += chunkSizeBytes;
+        mState  = (mProcessedBytes < mDerBytes.size()) ? State::kPrintBody : State::kPrintFooter;
+        hasLine = true;
+        break;
+    }
+    case State::kPrintFooter: {
+        // The `.50s` is to make sure the header is not wider than out internal string buffer. We clamp the footer.
+        mStringBuilder.Reset().AddFormat("-----END %.50s-----", mEncodedElement);
+        mState  = State::kDone;
+        hasLine = true;
+        break;
+    }
+    case State::kDone:
+        [[fallthrough]];
+    default: {
+        // Default initialized StringBuilder: empty output.
+        mState  = State::kDone;
+        hasLine = false;
+        break;
+    }
+    }
+
```

Comment:
nit: could `VerifyOrDie(mStringBuilder.Fit());` as we assume our lines are ok and we never check string builder retunrs (because we know they should fit...)